specs/sharding/beacon-chain.md

```diff
@@ -103,460 +101,342 @@ The following values are (non-configurable) constants used throughout the specif
-
-| Name | Value |
-| - | - |
-| `TIMELY_SHARD_WEIGHT` | `uint64(8)` |
+With the introduction of intermediate blocks the number of slots per epoch is doubled (it counts beacon blocks and intermediate blocks).
 
-TODO: `WEIGHT_DENOMINATOR` needs to be adjusted, but this breaks a lot of Altair code.
-
-## Preset
-
-### Misc
-
-| Name | Value | Notes |
-| - | - | - |
-| `MAX_SHARDS` | `uint64(2**10)` (= 1,024) | Theoretical max shard count (used to determine data structure sizes) |
-| `INITIAL_ACTIVE_SHARDS` | `uint64(2**6)` (= 64) | Initial shard count |
-| `SAMPLE_PRICE_ADJUSTMENT_COEFFICIENT` | `uint64(2**3)` (= 8) | Sample price may decrease/increase by at most exp(1 / this value) *per epoch* |
-| `MAX_SHARD_PROPOSER_SLASHINGS` | `2**4` (= 16) | Maximum amount of shard proposer slashing operations per block |
-| `MAX_SHARD_HEADERS_PER_SHARD` | `4` | |
-| `SHARD_STATE_MEMORY_SLOTS` | `uint64(2**8)` (= 256) | Number of slots for which shard commitments and confirmation status is directly available in the state |
-| `BLOB_BUILDER_REGISTRY_LIMIT` | `uint64(2**40)` (= 1,099,511,627,776) | shard blob builders |
+| Name | Value | Unit | Duration |
+| - | - | :-: | :-: |
+| `SLOTS_PER_EPOCH` | `uint64(2**6)` (= 64) | slots | 8:32 minutes |
 
 ### Shard blob samples
 
 | Name | Value | Notes |
 | - | - | - |
-| `MAX_SAMPLES_PER_BLOB` | `uint64(2**11)` (= 2,048) | 248 * 2,048 = 507,904 bytes |
-| `TARGET_SAMPLES_PER_BLOB` | `uint64(2**10)` (= 1,024) | 248 * 1,024 = 253,952 bytes |
+| `SAMPLES_PER_BLOB` | `uint64(2**9)` (= 2,048) | 248 * 2,048 = 507,904 bytes |
 
 ### Precomputed size verification points
 
 | Name | Value |
 | - | - |
 | `G1_SETUP` | Type `List[G1]`. The G1-side trusted setup `[G, G*s, G*s**2....]`; note that the first point is the generator. |
 | `G2_SETUP` | Type `List[G2]`. The G2-side trusted setup `[G, G*s, G*s**2....]` |
-| `ROOT_OF_UNITY` | `pow(PRIMITIVE_ROOT_OF_UNITY, (MODULUS - 1) // int(MAX_SAMPLES_PER_BLOB * POINTS_PER_SAMPLE), MODULUS)` |
-
-### Gwei values
-
-| Name | Value | Unit | Description |
-| - | - | - | - |
-| `MAX_SAMPLE_PRICE` | `Gwei(2**33)` (= 8,589,934,592) | Gwei | Max sample charged for a TARGET-sized shard blob |  
-| `MIN_SAMPLE_PRICE` | `Gwei(2**3)` (= 8) | Gwei | Min sample price charged for a TARGET-sized shard blob |
+| `ROOT_OF_UNITY` | `pow(PRIMITIVE_ROOT_OF_UNITY, (MODULUS - 1) // int(SAMPLES_PER_BLOB * POINTS_PER_SAMPLE), MODULUS)` |
 
 ## Configuration
 
 Note: Some preset variables may become run-time configurable for testnets, but default to a preset while the spec is unstable.  
-E.g. `INITIAL_ACTIVE_SHARDS`, `MAX_SAMPLES_PER_BLOB` and `TARGET_SAMPLES_PER_BLOB`.
+E.g. `ACTIVE_SHARDS` and `SAMPLES_PER_BLOB`.
 
-## Updated containers
-
-The following containers have updated definitions to support Sharding.
+### Time parameters
 
-### `AttestationData`
+| Name | Value | Unit | Duration |
+| - | - | :-: | :-: |
+| `SECONDS_PER_SLOT` | `uint64(8)` | seconds | 8 seconds |
```

Comment:
Yay!! Looking forward to ethereum time becoming properly powers-of-two based :)

Reply:
The engineering effort to change the slot time is not insignificant - something to consider before settling on a value: because it started out as a constant, it has wiggled its way into many weak and soft dependencies, such as the timing between the various actions of validator clients, beacon nodes and so on, as well as third-party tooling that so far has relied on `slot = (now() - genesis) / 12`.

Reply:
This is a fair point. @vbuterin what are your thoughts on this?
It may also be safer to change this back to 12s, and adjust all constants (gas limit etc.) accordingly to the new paradigm which only has a payload every 24s.

Reply:
I do feel like 24 seconds is too long for a block time and risks harming usability, and 6 seconds might be too short for a slot, so splitting the pain 50/50 (by making beacon slots 8s and user-experienced slots 16s) really is the best path by a significant margin.